pgjdbc/src/main/java/org/postgresql/jdbc/PgPreparedStatement.java

```diff
@@ -1589,16 +1577,23 @@ public void setCharacterStream(@Positive int parameterIndex,
   public void setBinaryStream(@Positive int parameterIndex, @Nullable InputStream value,
       @NonNegative @IntRange(from = 0, to = Integer.MAX_VALUE) long length)
       throws SQLException {
+    checkClosed();
+
+    if (value == null) {
+      setNull(parameterIndex, Oid.BYTEA);
```

Comment:
this was setNull(parameterIndex, Types.VARBINARY); in the first method

Reply:
I see, but in method setNull(int parameterIndex, int sqlType), “Types.VARBINARY” and “oid = Oid.BYTEA” are equivalent.
```java
case Types.BINARY:
case Types.VARBINARY:
case Types.LONGVARBINARY:
   oid = Oid.BYTEA;
   break;
```

Reply:
Good catch, I'll withdraw my criticism

Reply:
glad to help with the improvement of PG

Reply:
This should be a constant from `java.sql.Types`, not an `Oid`.
So `Oid.BYTEA` is not valid here, and it causes tests to fail.